msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/common/algorithms/Peak3DTest.java

```diff
@@ -0,0 +1,285 @@
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1 as published by the Free
+ * Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by the Eclipse Foundation.
+ */
+package io.github.msdk.featdet.ADAP3D.common.algorithms;
+
+import org.apache.commons.collections4.map.MultiKeyMap;
+
+import java.lang.Math;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * <p>
+ * Peak3DTest is used for testing a peak by comparing adjacent m/z-slices in profile data.
+ * </p>
+ */
+public class Peak3DTest {
+	
+	/**
+	 * <p>
+	 * Result class is used for returning lower and upper mz bound,boolean good peak value and list of similarity value.
+	 * </p>
+	 */
+	//Object of this class will return lowest mz boundary and highest mz boundary of adjacent similar peaks for given mz value.
+	//It will also return if the peak is good or not for given mz value.
+	public static class Result {
+		List<Double> similarityValues;
+		boolean goodPeak;
+		int lowerMzBound;
+		int upperMzBound;
+	}
+
+	enum Direction {UP, DOWN}
+
+	/* Instance of SliceSparseMatrix containing the profile data */
+	private final SliceSparseMatrix objsliceSparseMatrix;
+
+	/* Full-Width Half-Max of m/z-profiles */
+	private final double fwhm;
+
+	public Peak3DTest(SliceSparseMatrix objsliceSparseMatrix, double fwhm) {
+	    this.objsliceSparseMatrix = objsliceSparseMatrix;
+	    this.fwhm = fwhm;
+    }
+
+	/**
+	 * <p>
+	 * execute method is used for testing a peak with given m/z-value (variable mz)
+     * and left and right bounds (variables leftBound and rightBound).
+     *
+     * Peak is tested by comparing similarity between adjacent m/z slices.
+     *
+     * Let mzValues be a sorted list of all m/z values in the profile data. Let index be an integer such that
+     *
+     *    mzValues[index] == mz
+     *
+     * We find similarities between the EIC corresponding to m/z value mzValues[index] and adjacent EICs
+     * corresponding to
+     *
+     *    ..., mzValues[index-2], mzValues[index-1], mzValues[index+1], mzValues[index+2], ...
+     *
+     * as long as those similarities are higher than the similarity threshold. First, we check each m/z-value higher
+     * than mzValues[index], stop when the current similarity becomes lower than the similarity threshold, and save the last
+     * m/z-value (variable upperMZbound). Next, we check each m/z-value lower than mzValues[index], stop when the
+     * current similarity becomes lower than the similarity threshold, and save the last m/z-value (variable lowerMZbound)
+     *
+     * Peak is considered to be good if the differences
+     *    upperMZbound - mzValues[index],
+     *    mzValues[index] - lowerMZbound,
+     *    upperMZbound - lowerMZbound
+     * exceed certain thresholds, which depend on FWHM-value.
+     *
+	 *
+	 * @param mz a {@link java.lang.Double} object.
```

Comment:
I have written like this because I saw same comment style for other MSDK classes. I can write description if you want.